### PR TITLE
Narrow no_std_fs_operations exemptions with Whitaker module-path suppressions (#462)

### LINKED ISSUE #462: Narrow no_std_fs_operations exemptions with Whitaker module-path suppressions
## Summary

Replace Netsuke's broad `no_std_fs_operations` crate exemptions with Whitaker's
new module-path suppression wherever ambient filesystem access is confined to a
bounded module.

Whitaker PR [#315](https://github.com/leynos/whitaker/pull/315) adds
`excluded_paths` entries such as:

```toml
[no_std_fs_operations]
excluded_paths = ["netsuke::stdlib::which::lookup"]
```

Entries are anchored at the Rust crate identifier and match that module and its
descendants on segment boundaries.

## Current state

`dylint.toml` excludes the complete `netsuke`, `build_script_build`, and
`test_support` crates, plus many integration-test crates. This prevents the lint
from enforcing capability-oriented filesystem access in unaffected production
modules.

Whitaker v0.2.7 predates PR #315, so this work first needs a Whitaker release
that contains merge commit `dbc930a1dd74381b255a2beee6d6a43ab7780da8`.

## Proposed work

- Upgrade `whitaker-installer` and the installed lint suite to the first release
  containing Whitaker PR #315.
- Run the lint without the broad `netsuke` exclusion and inventory every
  resulting `std::fs` diagnostic.
- Replace the crate-wide exemption with the smallest justified
  `excluded_paths` set for genuine application-boundary operations, including
  any executable probing, cross-directory canonicalization, or durable-file
  synchronization that cannot be expressed through an existing capability.
- Reassess `build_script_build`, `test_support`, and integration-test crate
  exclusions. Keep crate-wide exclusions only where ambient filesystem access
  is genuinely the crate's boundary; otherwise narrow them to module paths.
- Update `docs/whitaker-users-guide.md`, `docs/developers-guide.md`, and the
  comments in `dylint.toml` with the ownership boundary and reuse policy for
  every retained exclusion.

## Acceptance criteria

- [ ] The Whitaker version used locally and in CI contains `excluded_paths`
      support from Whitaker PR #315.
- [ ] The whole `netsuke` crate is no longer listed under `excluded_crates`.
- [ ] Every production `excluded_paths` entry names a bounded module, explains
      why ambient access is required, and does not exempt an unrelated sibling.
- [ ] All remaining crate-wide exclusions are reviewed and justified in
      `dylint.toml` and contributor documentation.
- [ ] A negative check demonstrates that an unexcluded `std::fs` use in the
      same crate is still reported.
- [ ] `make check-fmt`, `make lint`, `make test`, and `make markdownlint` pass.

## References

- Whitaker PR [#315](https://github.com/leynos/whitaker/pull/315)
- `dylint.toml`
- `docs/whitaker-users-guide.md`
- `docs/developers-guide.md`

### PR DESCRIPTION
**Pull request by leynos**
Closes #462.

Whitaker PR [#315](https://github.com/leynos/whitaker/pull/315) adds
`excluded_paths` to `no_std_fs_operations`, letting an exemption name a bounded
module instead of a whole crate. This change takes Netsuke off the crate-wide
`netsuke` exemption so the capability-oriented filesystem policy is actually
enforced across the application crate.

## What changed

- **Whitaker upgrade.** `docs/whitaker-users-guide.md` is refreshed from
  upstream `main` (it documents `excluded_paths`), and the CI installer pin
  moves from `0.2.6` to `0.2.7`. The lint libraries the installer stages come from the
  `rolling` release built at merge commit
  `dbc930a1dd74381b255a2beee6d6a43ab7780da8`, which is the commit that carries
  PR #315.

- **`netsuke` is no longer in `excluded_crates`.** Running the lint with all
  exclusions cleared produced exactly ten production diagnostics, in two
  places. Both are now named as `excluded_paths`:

  - `netsuke::stdlib::which::lookup` — PATH resolution for the `which` standard
    library function. Candidate directories come from the ambient `PATH`
    variable, so executable metadata probing and cross-directory
    canonicalization cannot be scoped to a caller-supplied `cap_std::fs::Dir`.
    The entry covers the module and its `workspace` submodule (nine
    diagnostics) and reaches no sibling `which` module.
  - `netsuke::runner::process::file_io::ambient_sync` — the remaining
    diagnostic was a single `sync_all` on a `tempfile` handle. Rather than
    exempt the whole of `file_io`, which is otherwise capability-based, that
    one function moves into a new bounded `ambient_sync` submodule and only
    that module is exempt.

- **Crate-wide exclusions reassessed.** `build_script_build` is retained with a
  stated justification; each integration test crate in the list was confirmed
  to still emit diagnostics without its exclusion, and the two stale entries
  that did not — `dependabot_config_tests` and `logging_stderr_tests` — are
  removed.

- **`test_support` is now actually gated.** The crate is excluded from the root
  Cargo workspace, so a workspace-root `cargo dylint` never reached it and its
  root `dylint.toml` entry was inert. `make lint-whitaker` now runs the suite a
  second time from `test_support/`, against a `test_support/dylint.toml` whose
  only exemption is `test_support::fs` — the crate's ambient fixture boundary.
  Clearing the backlog that surfaced:

  - `check_ninja`, `exec`, `manifest`, and the crate root now route through
    `crate::fs` rather than calling `std::fs` directly.
  - The `original_dir` rstest fixture is fallible instead of expecting inside
    itself, and the localiser test lock recovers from poisoning the way
    `env_lock` already does. Both are fixtures, which the lint cannot recognize
    as test code.
  - The `assert_underlying_lock_is_*` helpers became macros, so a failure
    reports the calling test's line number.
  - `http`'s test module moved to `http_tests.rs`, bringing the module back
    under `module_max_lines`.
  - Missing inner doc comments added; an unused `anyhow` import dropped.

- **Documentation.** `dylint.toml` and `docs/developers-guide.md` now carry the
  ownership boundary for every retained exclusion plus an explicit reuse
  policy: thread a `cap_std::fs::Dir` to the call site or route through an
  existing boundary module rather than adding an entry.

## Negative check

A temporary `std::fs::metadata` call was added in two unexcluded locations and
the lint was re-run:

- `src/stdlib/which/cache.rs` — a sibling of the excluded `lookup` module
- the body of `src/runner/process/file_io.rs`, outside `ambient_sync`

Both were reported as errors, confirming the exclusions leak neither to
siblings nor to the parent module. The same check was run for `test_support`: a
`std::fs::read` added to `test_support/src/exec.rs` was reported despite
`test_support::fs` being exempt. All probes were reverted. The procedure is
recorded in `docs/developers-guide.md` so it can be repeated.

Separately, an in-source `#[allow(no_std_fs_operations)]` was tested and is
*not* honoured by the pinned build, so the existing guidance to configure
exemptions through `dylint.toml` still holds.

## Verification

`make check-fmt`, `make lint`, `make test`, `make markdownlint`, and
`make nixie` all pass. `coderabbit review --agent` reports no findings.

## References

- Lody session: https://lody.ai/leynos/sessions/dd2e1328-91ff-4c9a-83b6-965934b3f187

### CODE REVIEW

**Comment by pandalump** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai resume

File: docs/developers-guide.md [hunk 1]
```diff
@@ -90,15 +90,65 @@ Run these commands before finalizing any change:
...
-enumerated integration-test and workflow-contract crates. Netsuke itself needs
-ambient access for executable discovery through `PATH`, cross-directory symlink
-canonicalization, and temporary-file synchronization; other filesystem access
-should remain capability-scoped.
+Whitaker is configured by `dylint.toml` at the repository root.
+`docs/whitaker-users-guide.md` is a verbatim import of the upstream guide;
+refresh it from
+`https://raw.githubusercontent.com/leynos/whitaker/refs/heads/main/docs/users-guide.md`
+rather than editing it in place, and record Netsuke-specific policy here and in
+`dylint.toml`.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/developers-guide.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/whitaker-users-guide.md [hunk 2]
```diff
@@ -629,8 +630,22 @@ operations without triggering diagnostics. This is useful for:
...
+needs ambient filesystem access, while the rest stays under the capability
+policy.
+
+> **Note:** For both options, use Rust identifiers (underscores), not Cargo
+> package names (hyphens). For example, use `my_cli_app` rather than
+> `my-cli-app`, and `my_app::legacy_io` rather than `my-app::legacy_io`.
+>
+> **Tip:** For an ad hoc, single-site exemption that travels with the code, a
+> standard `#[allow(no_std_fs_operations)]` attribute on the item or module
+> also works, since the lint honours Rust's lint-level attributes.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/whitaker-users-guide.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: test_support/src/cwd_guard.rs [hunk 3]
```diff
@@ -39,18 +41,23 @@ mod tests {
...
 
     #[rstest]
     #[case(CwdGuard::acquire)]
     #[case(CwdGuard::new)]
     fn constructor_captures_current_directory(
-        original_dir: std::path::PathBuf,
+        original_dir: io::Result<std::path::PathBuf>,
         #[case] ctor: fn() -> io::Result<CwdGuard>,
     ) {
+        let original_dir = original_dir.expect("current_dir");
```
[the remark this answers is not in the document]
Comment (leynos) [reply test_support/src/cwd_guard.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: test_support/src/env_lock.rs [hunk 4]
```diff
@@ -64,15 +64,26 @@ impl Drop for EnvLock {
...
-    fn assert_underlying_lock_is_released(message: &str) {
-        let lock = ENV_LOCK.try_lock().expect(message);
-        drop(lock);
+    macro_rules! assert_underlying_lock_is_released {
+        ($message:expr $(,)?) => {
+            match ENV_LOCK.try_lock() {
+                Ok(lock) => drop(lock),
+                Err(err) => panic!("{}: {err}", $message),
+            }
+        };
```
[the remark this answers is not in the document]
Comment (leynos) [reply test_support/src/env_lock.rs:135]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: test_support/src/localizer.rs [hunk 5]
```diff
@@ -45,8 +45,15 @@ pub struct EnLocalizer {
...
+    // A poisoned lock means an earlier test panicked while holding it. The lock
+    // guards nothing but the ordering of localiser installation, and
+    // `set_en_localizer` below re-establishes the global state unconditionally,
+    // so recovering the guard is safe. Panicking here would instead fail every
+    // subsequent test that takes this fixture. `crate::env_lock` recovers from
+    // poisoning the same way.
+    let lock = localizer_test_lock().unwrap_or_else(PoisonError::into_inner);
     EnLocalizer {
-        _lock: localizer_test_lock().expect("localizer test lock poisoned"),
+        _lock: lock,
```
[the remark this answers is not in the document]
Comment (leynos) [reply test_support/src/localizer.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: test_support/src/exec.rs [hunk 6]
```diff
@@ -15,14 +15,9 @@
...
+use crate::fs;
 use anyhow::{Context, Result};
-use std::{
-    fs,
-    path::{Path, PathBuf},
-};
-
-#[cfg(unix)]
-use std::os::unix::fs::PermissionsExt;
+use std::path::{Path, PathBuf};
```
[the remark this answers is not in the document]
Comment (leynos) [reply test_support/src/exec.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: test_support/src/http_tests.rs [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        })
+        .unwrap_or_else(|| format!("{panic_payload:?}"));
+    assert!(
+        panic_text.contains(&format!("accept_timeout={:?}", accept_timeout)),
+        "panic message should embed the accept timeout: {panic_text}",
+    );
+    assert!(
+        panic_text.contains(&format!("poll_interval={:?}", poll_interval)),
+        "panic message should embed the poll interval: {panic_text}",
+    );
```
[the remark this answers is not in the document]
Comment (leynos) [reply test_support/src/http_tests.rs:165]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by pandalump** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai re:

> Add focused compile-fail/UI tests, or an equivalent lint harness, that assert allowed and unexcluded `std::fs` uses with stable diagnostic expectations.

It is not thr responsibility of this repository's developer to test whitaker's capabilities. Assume that whitaker does what it says it does. If the developer encounters a bug, they will raise it with the whitaker repo.

File: tests/whitaker_boundary_contract.rs [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+fn excluded_paths_name_bounded_modules(#[case] relative: &str) -> Result<()> {
+    let paths = exclusion_list(&read_repo_file(Utf8Path::new(relative))?, "excluded_paths")?;
+    for path in &paths {
+        ensure!(
+            path.contains("::"),
+            "{relative}: {path:?} names a whole crate; excluded_paths entries \
+             must name a module so siblings stay under the capability policy"
+        );
+    }
+    Ok(())
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/whitaker_boundary_contract.rs:218]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai resume

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### Makefile (1)

`66-69`: _📐 Maintainability & Code Quality_ | _🔵 Trivial_ | _⚡ Quick win_
**Run and document tests for the excluded `test_support` crate.**
`test_support` is outside the root workspace, so the root Cargo commands do not discover its own test targets. This PR changes that crate's fixtures and test modules. Add explicit `nextest` and doctest commands for `test_support`, or verify that it has no runnable tests.
- `Makefile#L66-L69`: add test commands using `--manifest-path test_support/Cargo.toml`.
- `docs/developers-guide.md#L537-L539`: qualify “This runs every” as root-workspace coverage until the additional commands exist.
As per coding guidelines, changed Rust functionality must be validated with relevant unit and behavioural tests.

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@Makefile` around lines 66 - 69, The Makefile test targets do not cover the
excluded test_support crate. In Makefile lines 66-69, add explicit nextest and
doctest commands using --manifest-path test_support/Cargo.toml, or verify and
document that the crate has no runnable tests; in docs/developers-guide.md lines
537-539, qualify “This runs every” to state that it covers the root workspace
until these commands exist.

_Source: Coding guidelines_

**Comment by leynos** [comment]
@coderabbitai resume

**Comment by leynos** [comment]
@coderabbitai resume

**Comment by leynos** [comment]
@coderabbitai resume

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already. (Treat documentation and validation coverage as in scope).

### ❌ Failed checks (2 warnings, 4 inconclusive)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Developer Documentation | ⚠️ Warning | The guide documents Whitaker, workspace gates and executable helpers, but it omits new `test_support::fs::is_dir` and the changed `EnLocalizer` poison-recovery/drop-order contract. | Add developer-guide entries for `test_support::fs::is_dir` and the `EnLocalizer` lock, field-order and poison-recovery rules; link these from the test-isolation section. |
| Testing (Compile-Time / Ui) | ⚠️ Warning | The PR changes compile-time lint and macro/API behaviour, but adds no trybuild or equivalent compile-fail test; boundary tests only parse files and explicitly do not invoke Whitaker. | Add a focused compile-fail/UI harness for the changed lint or macro/API contracts, with stable diagnostic assertions or snapshots; keep runtime tests for other behaviour. |

File: test_support/dylint.toml [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+#
+# Where a `test_support::fs` wrapper does exist, prefer it inside these modules
+# too: the exemption is a floor for what the harness cannot express, not a
+# licence to stop using the boundary.
+excluded_paths = [
+    "test_support::fs",
+    "test_support::dev_fast::cargo_log",
+    "test_support::dev_fast::release",
+    "test_support::dev_fast::sandbox",
+    "test_support::dev_fast::staging",
```
[the remark this answers is not in the document]
Comment (leynos) [reply test_support/dylint.toml]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

[the remark this answers is not in the document]
Comment (leynos) [reply test_support/dylint.toml]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai resume

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already. (Treat documentation and validation coverage as in scope).

### ❌ Failed checks (2 errors, 2 warnings)

| Check name | Status | Explanation | Resolution |
| :---: | :--- | :--- | :--- |
| Testing (Overall) | ❌ Error | The suite leaves plausible regressions untested: `run_all` is never checked for the full invocation count, `env_lock` never exercises poisoned mutex states, and boundary properties test a duplicate... | Add an exact `run_all` invocation assertion, poison/recover `ENV_LOCK` tests, and an automated negative Whitaker probe or equivalent integration fixture for sibling and parent paths. |
| Security And Privacy | ❌ Error | Flag the new `TEST_SUPPORT_MANIFEST`: its documented caller-controlled value is expanded unquoted in three shell recipes, allowing metacharacters to inject commands. | Quote every expansion as `"$(TEST_SUPPORT_MANIFEST)"` and update the contract tests; preserve the documented override while preventing whitespace splitting and shell metacharacter injection. |
| Developer Documentation | ⚠️ Warning | The guide omits new `fs` helpers and `BuildScenario::run_all`, and it falsely describes `test_support` as having one exclusion; `test_support/dylint.toml` has four. | Document the new internal APIs and EnLocalizer lock/drop/poison contracts in `docs/developers-guide.md`, and list all `test_support` exclusions accurately. |
| Testing (Compile-Time / Ui) | ⚠️ Warning | Whitaker exclusions change compile-time lint diagnostics, but the contract tests explicitly do not invoke Whitaker and no Whitaker UI or compile-fail fixture exists. | Add a focused compile-fail/UI or equivalent harness test that asserts allowed boundary paths pass and unexcluded sibling or parent filesystem access emits the expected lint diagnostic. |

**Comment by leynos** [comment]
@coderabbitai re:

> Whitaker exclusions change compile-time lint diagnostics, but the contract tests explicitly do not invoke Whitaker and no Whitaker UI or compile-fail fixture exists.

Note that we should *not* be testing whitaker behaviour. It should be assumed to function as described.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### tests/dev_fast_make_target_tests.rs (1)

`50-82`: _🗄️ Data Integrity & Integration_ | _🟠 Major_ | _⚡ Quick win_
**Assert the expected Cargo invocation count.**
Store the result of `run_all` before iterating. Require one invocation for `dev-build` and two for `dev-test`. The current loop passes when `dev-test` records only the root invocation, so removing the `test_support` command can leave the test green. `BuildScenario::run_all` only guarantees a non-empty collection, so this test must enforce the target-specific count. ([raw.githubusercontent.com](https://raw.githubusercontent.com/leynos/netsuke/refs/pull/478/head/test_support/src/dev_fast/scenario.rs))

##### Suggested assertion

```diff
-    for invocation in scenario.run_all(target.name)? {
+    let invocations = scenario.run_all(target.name)?;
+    let expected_invocations = if target.name == "dev-test" { 2 } else { 1 };
+    ensure!(
+        invocations.len() == expected_invocations,
+        "`{}` should record {} Cargo invocation(s), got {}",
+        target.name,
+        expected_invocations,
+        invocations.len()
+    );
+    for invocation in invocations {
```

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@tests/dev_fast_make_target_tests.rs` around lines 50 - 82, Update the test
around BuildScenario::run_all to store its returned collection before iteration
and assert it contains one invocation for dev-build and two for dev-test. Keep
the existing per-invocation assertions unchanged, then iterate over the
validated collection so missing test_support invocations fail the test.

#### tests/makefile_test_target.rs (1)

`184-201`: _🎯 Functional Correctness_ | _🟠 Major_ | _⚡ Quick win_
**Make the manifest override shell-safe.**
The contract searches for the unquoted `--manifest-path $(TEST_SUPPORT_MANIFEST)` form, and the Makefile uses the same expansion. A value such as `tmp path/Cargo.toml` is split by the shell, so the advertised override fails. Export the variable and pass it as `--manifest-path "$${TEST_SUPPORT_MANIFEST}"`. Update this matcher and the scoped `RustflagsCase` markers, then add a path-with-spaces case. ([raw.githubusercontent.com](https://raw.githubusercontent.com/leynos/netsuke/refs/pull/478/head/Makefile))

##### Suggested contract shape

```diff
-    .filter(|line| line.contains("--manifest-path $(TEST_SUPPORT_MANIFEST)"))
+    .filter(|line| line.contains("--manifest-path \"$${TEST_SUPPORT_MANIFEST}\""))
```

Also applies to: 330-363

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@tests/makefile_test_target.rs` around lines 184 - 201, Update the Makefile
manifest override contract in behavioural_test_passes_also_target_test_support
and the related RustflagsCase markers to match the shell-safe --manifest-path
"$${TEST_SUPPORT_MANIFEST}" form. Ensure the Makefile exports
TEST_SUPPORT_MANIFEST and quotes the variable when invoking Cargo, then add a
test case using a manifest path containing spaces to verify the override works.

#### test_support/src/lib.rs (1)

`165-178`: _📐 Maintainability & Code Quality_ | _🔵 Trivial_ | _⚡ Quick win_
**Retain successful UTF-8-path coverage.**
Keep the invalid-UTF-8 regression case and retain a positive case for both fake-executable helpers. Assert that a normal UTF-8 sandbox produces usable executable paths through the sanctioned filesystem helper. Verify existing coverage first to avoid duplication.

```shell
#!/usr/bin/env bash
set -euo pipefail

rg -n -C 6 \
  'fake_ninja|fake_ninja_with_check|is_executable_file|fs::exists' \
  test_support tests
```

Based on the supplied change details, the previous successful creation and existence assertions were removed.  As per coding guidelines, changed Rust functionality must be validated with relevant unit and behavioural tests.

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@test_support/src/lib.rs` around lines 165 - 178, Restore or add positive
UTF-8 sandbox coverage for both fake_ninja and fake_ninja_check_build_file,
while preserving the non-UTF-8 rejection test. Assert each helper creates usable
executable paths and verify their existence through the project’s sanctioned
filesystem helper, reusing existing coverage rather than duplicating tests.

_Source: Coding guidelines_

File: tests/whitaker_boundary_contract.rs [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+                .split_once(':')
+                .is_none_or(|(name, rest)| name.trim() != target || rest.starts_with('='))
+    });
+    lines.next()?;
+    let recipe: Vec<&str> = lines
+        .take_while(|line| line.starts_with('\t') || line.trim().is_empty())
+        .filter(|line| line.starts_with('\t'))
+        .collect();
+    Some(recipe.join("\n"))
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/whitaker_boundary_contract.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai resume

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

#### tests/dev_fast_make_target_tests.rs (1)

`50-82`: _🗄️ Data Integrity & Integration_ | _🟠 Major_ | _⚡ Quick win_
**Assert the expected Cargo invocation count.**
Store the result of `run_all` before iterating. Require one invocation for `dev-build` and two for `dev-test`. The current loop passes when `dev-test` records only the root invocation, so removing the `test_support` command can leave the test green. `BuildScenario::run_all` only guarantees a non-empty collection, so this test must enforce the target-specific count. ([raw.githubusercontent.com](https://raw.githubusercontent.com/leynos/netsuke/refs/pull/478/head/test_support/src/dev_fast/scenario.rs))

##### Suggested assertion

```diff
-    for invocation in scenario.run_all(target.name)? {
+    let invocations = scenario.run_all(target.name)?;
+    let expected_invocations = if target.name == "dev-test" { 2 } else { 1 };
+    ensure!(
+        invocations.len() == expected_invocations,
+        "`{}` should record {} Cargo invocation(s), got {}",
+        target.name,
+        expected_invocations,
+        invocations.len()
+    );
+    for invocation in invocations {
```

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@tests/dev_fast_make_target_tests.rs` around lines 50 - 82, Update the test
around BuildScenario::run_all to store its returned collection before iteration
and assert it contains one invocation for dev-build and two for dev-test. Keep
the existing per-invocation assertions unchanged, then iterate over the
validated collection so missing test_support invocations fail the test.

#### tests/makefile_test_target.rs (1)

`184-201`: _🎯 Functional Correctness_ | _🟠 Major_ | _⚡ Quick win_
**Make the manifest override shell-safe.**
The contract searches for the unquoted `--manifest-path $(TEST_SUPPORT_MANIFEST)` form, and the Makefile uses the same expansion. A value such as `tmp path/Cargo.toml` is split by the shell, so the advertised override fails. Export the variable and pass it as `--manifest-path "$${TEST_SUPPORT_MANIFEST}"`. Update this matcher and the scoped `RustflagsCase` markers, then add a path-with-spaces case. ([raw.githubusercontent.com](https://raw.githubusercontent.com/leynos/netsuke/refs/pull/478/head/Makefile))

##### Suggested contract shape

```diff
-    .filter(|line| line.contains("--manifest-path $(TEST_SUPPORT_MANIFEST)"))
+    .filter(|line| line.contains("--manifest-path \"$${TEST_SUPPORT_MANIFEST}\""))
```

Also applies to: 330-363

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@tests/makefile_test_target.rs` around lines 184 - 201, Update the Makefile
manifest override contract in behavioural_test_passes_also_target_test_support
and the related RustflagsCase markers to match the shell-safe --manifest-path
"$${TEST_SUPPORT_MANIFEST}" form. Ensure the Makefile exports
TEST_SUPPORT_MANIFEST and quotes the variable when invoking Cargo, then add a
test case using a manifest path containing spaces to verify the override works.

#### test_support/src/lib.rs (1)

`165-178`: _📐 Maintainability & Code Quality_ | _🔵 Trivial_ | _⚡ Quick win_
**Retain successful UTF-8-path coverage.**
Keep the invalid-UTF-8 regression case and retain a positive case for both fake-executable helpers. Assert that a normal UTF-8 sandbox produces usable executable paths through the sanctioned filesystem helper. Verify existing coverage first to avoid duplication.

```shell
#!/usr/bin/env bash
set -euo pipefail

rg -n -C 6 \
  'fake_ninja|fake_ninja_with_check|is_executable_file|fs::exists' \
  test_support tests
```

Based on the supplied change details, the previous successful creation and existence assertions were removed.  As per coding guidelines, changed Rust functionality must be validated with relevant unit and behavioural tests.

##### 🤖 Detailed instructions

Verify each finding against current code. Fix only still-valid issues, skip the
rest with a brief reason, keep changes minimal, and validate.

In `@test_support/src/lib.rs` around lines 165 - 178, Restore or add positive
UTF-8 sandbox coverage for both fake_ninja and fake_ninja_check_build_file,
while preserving the non-UTF-8 rejection test. Assert each helper creates usable
executable paths and verify their existence through the project’s sanctioned
filesystem helper, reusing existing coverage rather than duplicating tests.

_Source: Coding guidelines_

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already. (Treat documentation and validation coverage as in scope).

### ❌ Failed checks (1 warning, 3 inconclusive)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Developer Documentation | ⚠️ Warning | The guide documents the fs boundary, is\_dir, and is\_executable\_file, but it does not document new fs::copy, fs::modified, or Unix fs::write\_with\_mtime APIs used by dev\_fast. | Add these helpers to the test\_support::fs section, including their purpose, error and Unix semantics, and usage in release and staging fixtures. |

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai resume

File: test_support/src/dev_fast/scenario.rs [hunk 11]
```diff
@@ -134,14 +134,33 @@ impl BuildScenario {
...
+
+    /// Run `target` and return every invocation it produced, in order.
+    pub fn run_all(&self, target: &str) -> Result<Vec<CargoInvocation>> {
+        self.run_recording(target)?;
+        let invocations = self.cargo.invocations()?;
+        ensure!(
+            !invocations.is_empty(),
+            "make {target} should invoke cargo at least once"
+        );
+        Ok(invocations)
```
[the remark this answers is not in the document]
Comment (leynos) [reply test_support/src/dev_fast/scenario.rs:189]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: docs/developers-guide.md [hunk 12]
```diff
@@ -138,33 +138,96 @@ configuration does and does not cover.
...
+`WHITAKER_INSTALLER_VERSION`.
+
+What the module-scoped exemptions in `dylint.toml` actually depend on is
+[Whitaker PR #315][whitaker-pr-315], which added the `excluded_paths` option,
+so the staged libraries must be recent enough to include it. Libraries staged
+from an older checkout ignore `excluded_paths` silently — the exemptions stop
+applying with no error, and the lint reports the modules they covered. Re-run
+`whitaker-installer` to restage from HEAD. If that checkout has been left on a
+detached HEAD, the install fails at its `git pull`; put it back on the default
+branch and re-run.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/developers-guide.md]: @coderabbitai the whitaker lint suite libraries are intentionally installed at HEAD (the whitaker installer is a distinct tool from the libraries)

**Comment by pandalump** [comment]
@coderabbitai review
